### PR TITLE
Refactor loc tests

### PR DESCRIPTION
**Pull request by Luni-4**
Thanks in advance for your review! :)

### CODE REVIEW

File: src/metrics/loc.rs [hunk 1]
```diff
@@ -480,28 +480,47 @@ mod tests {
...
+                      b,
+                      c):
+                 print(a)
+                 print(b)
+                 print(c)\n",
+            "foo.py",
+            PythonParser,
+            loc,
+            [
+                (sloc, 6, usize),
```
Comment (calixteman) [inline src/metrics/loc.rs:518]: Please add a comment for each value to explain why the value (exactly as you did for Halstead).

File: src/metrics/loc.rs [hunk 2]
```diff
@@ -377,20 +377,21 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_blank_python() {
+    fn test_blank() {
         check_metrics!(
-            "\na = 42\n\n",
+            "
+            a = 42
+            \n",
```
Comment (marco-c) [inline src/metrics/loc.rs:384]: Why are you removing `\n` sometimes and keeping it other times?
I'd prefer to be consistent and either always have `\n` or never have it.

File: src/metrics/loc.rs [hunk 3]
```diff
@@ -377,20 +377,21 @@ mod tests {
...
             "foo.py",
             PythonParser,
             loc,
             [(blank, 1, usize)]
         );
-    }
 
-    #[test]
-    fn test_blank_rust() {
```
Comment (marco-c) [inline src/metrics/loc.rs:391]: Maybe better to keep the tests as separated, so they can pass/fail individually

File: src/metrics/loc.rs [hunk 4]
```diff
@@ -480,28 +480,47 @@ mod tests {
...
                 break;\n",
             "foo.c",
             CppParser,
             loc,
             [(lloc, 2, usize)]
         );
     }
+
+    #[test]
+    fn test_general_loc() {
```
Comment (marco-c) [inline src/metrics/loc.rs:506]: Since this is a new test and is not part of the refactoring of the pre-existing tests, it should be part of a separate PR

File: src/metrics/loc.rs (hunk 3 above)
Comment (Luni-4) [reply src/metrics/loc.rs:391]: It's better to group tests on the same stuff in a single function, at least we immediately know the wrong metric. Having the same metric test for each language is a boilerplate to maintain and eventually update

**Comment by Luni-4** [comment]
Closing in favor of #239

File: src/metrics/loc.rs (hunk 3 above)
Comment (marco-c) [reply src/metrics/loc.rs:391]: If we keep them separated, we still immediately know the wrong metric, but we have the possibility to know if a specific language is broken.

Comment (Luni-4) [reply src/metrics/loc.rs:391]: I can't figure out the problem, sorry :)
Since each sub-test of a metric test is a macro, if a metric test is failing, we can immediately detect which macro is involved and at the same time we can determine if that specific language is broken.

Comment (marco-c) [reply src/metrics/loc.rs:391]: If there is a failure on the first assertion, the other assertions won't be evaluated.

In this case, if parsing Python fails, we won't parse Rust and C too, so we won't be able to tell right away if it's just a Python problem or if is a cross-language problem.

Comment (Luni-4) [reply src/metrics/loc.rs:391]: > If there is a failure on the first assertion, the other assertions won't be evaluated.
> 
> In this case, if parsing Python fails, we won't parse Rust and C too, so we won't be able to tell right away if it's just a Python problem or if is a cross-language problem.

Oh, so it's just a comfort matter. I thought to solve that problem commenting the series of sub-test macros until the botched one is found. Because if a test is failing we should get the hands on the same to discover the cause.

Actually I refactored the code because I didn't like all those little tests for the same thing

Comment (Luni-4) [reply src/metrics/loc.rs:391]: Anyway, I see your point, but I think it would be clear to have a test-per-metric instead of different test-per-metric-per-language as I thought initially

Comment (marco-c) [reply src/metrics/loc.rs:391]: Well, since there is no practical advantage in sticking the tests together, but there is an advantage in keeping them separated, let's keep them separated.

Comment (Luni-4) [reply src/metrics/loc.rs:391]: I don't like this approach, but if it is more comfortable for you, let's keep them as I thought initially :P
